### PR TITLE
Add `--experimental.virtual-machine.mount.9p.*` settings

### PR DESCRIPTION
**Pull request by jandubois**
This commit temporarily exports kubeBackend.k3sHelper so that we can call k3sHelper.requiresRestartReasons from the Lima backend.

All the requiresRestartReasons code should be moved to the BackendHelper instead.

### CODE REVIEW

**Comment by jandubois** [comment]
Created #4055 to revert making `k3sHelper` public.

**Review by mook-as — CHANGES_REQUESTED** [review]

File: pkg/rancher-desktop/backend/k8s.ts [hunk 1]
```diff
@@ -125,6 +125,8 @@ export interface KubernetesBackend extends EventEmitter<KubernetesBackendEvents>
    * given new configuration been applied on top of the existing old configuration.
    */
   requiresRestartReasons(oldConfig: BackendSettings, newConfig: RecursivePartial<BackendSettings>, extras?: ExtraRequiresReasons): Promise<RestartReasons>;
+
+  readonly k3sHelper: K3sHelper;
```
[the remark this answers is not in the document]
Comment (jandubois) [reply pkg/rancher-desktop/backend/k8s.ts:129]: See above:

> Created https://github.com/rancher-sandbox/rancher-desktop/issues/4055 to revert making k3sHelper public.

File: pkg/rancher-desktop/main/commandServer/settingsValidator.ts [hunk 2]
```diff
@@ -219,14 +242,30 @@ export default class SettingsValidator {
...
+        if (mergedSettings.experimental.virtualMachine.mount.type !== MountType.NINEP) {
+          errors.push(`Field ${ fqname } can only be changed when experimental.virtualMachine.mount.type is "${ MountType.NINEP }".`);
+
+          return false;
+        }
+      }
+
+      // any changed settings must be valid, even if we ignore them because the mount.type is not 9p
+      return validator.call(this, mergedSettings, currentValue, desiredValue, errors, fqname) &&
+        mergedSettings.experimental.virtualMachine.mount.type === MountType.NINEP;
```
[the remark this answers is not in the document]
Comment (jandubois) [reply pkg/rancher-desktop/main/commandServer/settingsValidator.ts]: No, the second half was based on a thinko; removed.

File: pkg/rancher-desktop/config/settings.ts [hunk 3]
```diff
@@ -30,6 +30,10 @@ export enum ContainerEngine {
   MOBY = 'moby',
 }
 
+export enum MountType {
```
[the remark this answers is not in the document]
Comment (jandubois) [reply pkg/rancher-desktop/config/settings.ts]: I don't think so, it is just for convenient that the values for `MountType` match the `mountType` value in `lima.yaml`. But the type is used in our own `Settings` structure.

[the remark this answers is not in the document]
Comment (mook-as) [reply pkg/rancher-desktop/config/settings.ts]: Right, but those settings never make sense outside of lima, right? (In the current case, they don't make sense Windows; and ages ago, they don't make sense on minikube.)

[the remark this answers is not in the document]
Comment (jandubois) [reply pkg/rancher-desktop/config/settings.ts]: Yeah, but if we ever have types that make sense on Windows, we would just add them. Adding them to a `LimaMountType` wouldn't make sense.

It is the job of the validator to filter out the settings that are not applicable to the current platform/architecture etc, but the types themselves should be a superset of all settings valid on any of the platforms.

As long as the fieldname is `virtualMachine.mount.type` I think the typename should be `MountType`. 

If the type was `LimaMountType` then I would expect the field name to be `mount.limaMountType` which means you would have to set it with `--experimental.virtual-machine.mount.lima-mount-type`, which I don't want.

**Review by mook-as — APPROVED** [review]
Looks fine, seems to work.

**Pull request merged**